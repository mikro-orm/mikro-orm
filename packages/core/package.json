{
  "name": "@mikro-orm/core",
  "version": "6.1.12",
  "description": "TypeScript ORM for Node.js based on Data Mapper, Unit of Work and Identity Map patterns. Supports MongoDB, MySQL, PostgreSQL and SQLite databases as well as usage with vanilla JavaScript.",
  "main": "dist/index.js",
  "module": "dist/index.mjs",
  "typings": "dist/index.d.ts",
  "exports": {
    "./package.json": "./package.json",
    ".": {
      "import": {
        "types": "./dist/index.d.ts",
        "default": "./dist/index.mjs"
      },
      "require": "./dist/index.js"
    }
  },
  "repository": {
    "type": "git",
    "url": "git+ssh://git@github.com/mikro-orm/mikro-orm.git"
  },
  "funding": "https://github.com/sponsors/b4nan",
  "keywords": [
    "orm",
    "mongo",
    "mongodb",
    "mysql",
    "mariadb",
    "postgresql",
    "sqlite",
    "sqlite3",
    "ts",
    "typescript",
    "js",
    "javascript",
    "entity",
    "ddd",
    "mikro-orm",
    "unit-of-work",
    "data-mapper",
    "identity-map"
  ],
  "author": "Martin AdÃ¡mek",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/mikro-orm/mikro-orm/issues"
  },
  "homepage": "https://mikro-orm.io",
  "engines": {
    "node": ">= 18.12.0"
  },
  "scripts": {
    "build": "yarn clean && yarn compile && yarn copy && yarn run -T gen-esm-wrapper dist/index.js dist/index.mjs",
    "clean": "yarn run -T rimraf ./dist",
    "compile": "yarn run -T tsc -p tsconfig.build.json",
    "copy": "node ../../scripts/copy.mjs"
  },
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "dataloader": "2.2.2",
    "dotenv": "16.4.5",
    "esprima": "4.0.1",
    "fs-extra": "11.2.0",
    "globby": "11.1.0",
    "mikro-orm": "6.1.12",
    "reflect-metadata": "0.2.2"
  }
}
