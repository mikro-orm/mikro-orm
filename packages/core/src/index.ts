/**
 * @packageDocumentation
 * @module core
 */
/* istanbul ignore file */
export {
  Constructor, Dictionary, PrimaryKeyType, PrimaryKeyProp, Primary, IPrimaryKey, FilterQuery, IWrappedEntity, EntityName, EntityData, Highlighter,
  AnyEntity, EntityProperty, EntityMetadata, QBFilterQuery, PopulateOptions, Populate, Loaded, New, LoadedReference, LoadedCollection,
  GetRepository, EntityRepositoryType, MigrationObject, DeepPartial, PrimaryProperty, Cast, IsUnknown, EntityDictionary, EntityDTO, PlainObject,
} from './typings';
export * from './enums';
export * from './errors';
export * from './exceptions';
export * from './MikroORM';
export * from './entity';
export * from './events';
export * from './EntityManager';
export * from './unit-of-work';
export * from './utils';
export * from './hydration';
export * from './drivers';
export * from './connections';
export * from './platforms';
export * from './types';
export * from './naming-strategy';
export * from './metadata';
export * from './cache';
export * from './decorators';
