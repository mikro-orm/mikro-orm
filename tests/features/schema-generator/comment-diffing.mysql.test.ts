import { Entity, MikroORM, PrimaryKey, Property } from '@mikro-orm/core';
import type { PostgreSqlDriver, SchemaGenerator } from '@mikro-orm/postgresql';

@Entity({ tableName: 'book' })
export class Book0 {

  @PrimaryKey()
  id!: number;

  @Property()
  name!: string;

}

@Entity({ tableName: 'book', comment: 'this is book\'s table' })
export class Book1 {

  @PrimaryKey({ comment: 'this is primary\'s key' })
  id!: number;

  @Property({ comment: 'this is name of book' })
  name!: string;

}

@Entity({ tableName: 'book', comment: 'table comment' })
export class Book2 {

  @PrimaryKey({ comment: 'new comment' })
  id!: number;

  @Property({ comment: '' })
  name!: string;

}

@Entity({ tableName: 'book' })
export class Book3 {

  @PrimaryKey()
  id!: number;

  @Property()
  name!: string;

}

describe('comment diffing in mysql', () => {

  let orm: MikroORM<PostgreSqlDriver>;
  let generator: SchemaGenerator;

  beforeAll(async () => {
    orm = await MikroORM.init({
      entities: [Book0],
      dbName: `mikro_orm_test_comments`,
      type: 'mysql',
      port: 3308,
    });
    generator = orm.getSchemaGenerator();
    await generator.ensureDatabase();
    await generator.execute('drop table if exists book');
    await generator.createSchema();
  });

  afterAll(() => orm.close(true));

  test('schema generator updates comments', async () => {
    await orm.discoverEntity(Book1);
    orm.getMetadata().reset('Book0');
    const diff1 = await generator.getUpdateSchemaSQL({ wrap: false });
    expect(diff1).toMatchSnapshot();
    await generator.execute(diff1);

    orm.getMetadata().reset('Book1');
    await orm.discoverEntity(Book2);
    const diff2 = await generator.getUpdateSchemaSQL({ wrap: false });
    expect(diff2).toMatchSnapshot();
    await generator.execute(diff2);

    orm.getMetadata().reset('Book2');
    await orm.discoverEntity(Book3);
    const diff3 = await generator.getUpdateSchemaSQL({ wrap: false });
    expect(diff3).toMatchSnapshot();
    await generator.execute(diff3);

    await expect(generator.getUpdateSchemaSQL({ wrap: false })).resolves.toBe('');
  });

});
