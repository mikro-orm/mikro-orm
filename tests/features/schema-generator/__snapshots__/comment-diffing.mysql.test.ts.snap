// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comment diffing in mysql schema orm.schema updates comments 1`] = `
"alter table \`book\` modify \`id\` int unsigned auto_increment not null comment 'this is primary\\'s key';
alter table \`book\` modify \`name\` varchar(255) not null comment 'this is name of book';
alter table \`book\` comment = 'this is book\\'s table';

"
`;

exports[`comment diffing in mysql schema orm.schema updates comments 2`] = `
"alter table \`book\` modify \`id\` int unsigned auto_increment not null comment 'new comment';
alter table \`book\` modify \`name\` varchar(255) not null;
alter table \`book\` comment = 'table comment';

"
`;

exports[`comment diffing in mysql schema orm.schema updates comments 3`] = `
"alter table \`book\` modify \`id\` int unsigned auto_increment not null;
alter table \`book\` comment = '';

"
`;
